src/logsumexp.jl

```diff
@@ -61,36 +61,59 @@ _logsumexp_onepass_reduce(X, ::Base.EltypeUnknown) = reduce(_logsumexp_onepass_o
 ## Reductions for one-pass algorithm: avoid expensive multiplications if numbers are reduced
 
 # reduce two numbers
-function _logsumexp_onepass_op(x1, x2)
-    a = x1 == x2 ? zero(x1 - x2) : -abs(x1 - x2)
-    xmax = x1 > x2 ? oftype(a, x1) : oftype(a, x2)
```

Comment:
So the `oftype` isn't needed now?

Reply:
No, it is not needed anymore since the terms in both branches are guaranteed to be of the same type since the relevant arguments are ensured to be of the same type. I thought in this way the code would be easier to write and to read.

Reply:
But `oftype(x1, NaN)` may not be of the same type as `x1 - x2`, right? This should be `oftype(x1-x2, NaN)`.

Reply:
`x1` and `x2` are of the same type, so `oftype(x1, NaN)` should be of the same type as `x1 - x2`.